File: hono_bk_end/src/logger.ts
import { appendFile, mkdir } from "node:fs/promises";
import { dirname, resolve } from "node:path";

const LOG_FILE = resolve(process.cwd(), "logs", "patch.log");

export async function logPatchEvent(
	event: string,
	data: Record<string, unknown> = {},
): Promise<void> {
	const line = JSON.stringify({
		time: new Date().toISOString(),
		event,
		...data,
	});

	console.log(line);
	await mkdir(dirname(LOG_FILE), { recursive: true });
	await appendFile(LOG_FILE, `${line}\n`, "utf8");
}

File: nanobot_bk_end/src/logger.py
"""JSONL event logger matching the legacy Hono backend format."""

from __future__ import annotations

import json
import os
from datetime import datetime, timezone
from pathlib import Path
from typing import Any


LOG_FILE = Path(os.environ.get("LOG_FILE_PATH", "logs/patch.log")).resolve()


def log_event(event: str, data: dict[str, Any] | None = None) -> None:
    """Append a JSON event line to the log file.

    Mirrors the legacy logger.ts format:
    { "time": "...", "event": "...", ...data }\n\n
    """
    line = json.dumps(
        {
            "time": datetime.now(timezone.utc).isoformat(),
            "event": event,
            **(data or {}),
        },
        ensure_ascii=False,
    )
    print(line)
    LOG_FILE.parent.mkdir(parents=True, exist_ok=True)
    with LOG_FILE.open("a", encoding="utf-8") as f:
        f.write(f"{line}\n\n")
